controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -52,41 +52,54 @@ beerocks::eChannelScanErrCode dynamic_channel_selection_task::dcs_request_scan_t
 {
     // When a scan is requested send the scan parameters Channel pool & Dwell time
 
+    auto radio_mac_str = beerocks::net::network_utils::mac_to_string(m_radio_mac);
+
+    //Create request
     auto request = beerocks::message_com::create_vs_message<
         beerocks_message::cACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_REQUEST>(cmdu_tx);
     if (!request) {
         LOG(ERROR) << "Failed building message cACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_REQUEST";
         return beerocks::eChannelScanErrCode::CHANNEL_SCAN_INTERNAL_FAILURE;
     }
+    request->scan_params().radio_mac = m_radio_mac;
 
+    //Get dwell time, if dwell time invalid fail scan.
     int32_t dwell_time_msec = database.get_channel_scan_dwell_time_msec(m_radio_mac, m_single_scan);
     if (dwell_time_msec <= 0) {
         LOG(ERROR) << "invalid dwell_time <= 0";
         return beerocks::eChannelScanErrCode::CHANNEL_SCAN_INVALID_PARAMS;
     }
+    request->scan_params().dwell_time_ms = dwell_time_msec;
 
-    //get current channel pool from DB
+    //Get channel pool, if channel pool empty fil scan, if pool is set to "all channel" use
+    //supported channels instead.
     auto &curr_channel_pool = database.get_channel_scan_pool(m_radio_mac, m_single_scan);
-    if (curr_channel_pool.empty()) {
-        LOG(ERROR) << "empty channel pool is not supported. please set channel pool for mac="
-                   << m_radio_mac;
-        return beerocks::eChannelScanErrCode::CHANNEL_SCAN_INVALID_PARAMS;
-    }
-
-    if (curr_channel_pool.size() > beerocks::message::SUPPORTED_CHANNELS_LENGTH) {
-        LOG(ERROR) << "channel_pool is too big";
-        return beerocks::eChannelScanErrCode::CHANNEL_SCAN_POOL_TOO_BIG;
+    if (curr_channel_pool.size() == 1 && (*curr_channel_pool.begin()) == 0) {
+        LOG(DEBUG) << "Using all supported channels for channel scan on mac=" << m_radio_mac;
+        auto supported_channels = database.get_hostap_supported_channels(radio_mac_str);
+        request->scan_params().channel_pool_size = supported_channels.size();
+        std::transform(supported_channels.begin(), supported_channels.end(),
+                       request->scan_params().channel_pool,
+                       [](beerocks::message::sWifiChannel &c) -> uint8_t { return c.channel; });
+    } else {
+        LOG(DEBUG) << "Using custom set channels for channel scan on mac=" << m_radio_mac;
+        if (curr_channel_pool.empty()) {
+            LOG(ERROR) << "empty channel pool is not supported. please set channel pool for mac="
+                       << m_radio_mac;
+            return beerocks::eChannelScanErrCode::CHANNEL_SCAN_INVALID_PARAMS;
```

Comment:
I still don't understand why the empty channel pool can't be used as the marker... In this case, it's an error of the pool is empty. And the only other case where the channel pool is requested is to construct cACTION_BML_CHANNEL_SCAN_GET_CONTINUOUS_PARAMS_RESPONSE - whether we report an empty pool there or a pool with just zero shouldn't make a difference, should it?

Anyway I'll trust your judgement on this.

Reply:
I've commented in the Linked Issue #1216
But basically the reason is that, currently on initialization the channel pool starts empty.
If an empty pool means "All Channels" then when we request a scan before setting the channel pool, the scan will trigger as if the user requested a scan on all channels since the initial pool is empty.
This is incorrect as we want the user to actively choose if they want to scan on all channels by setting it to a designated identifier, and fail to scan if no channel pool is set.